common/lib/xmodule/xmodule/mongo_connection.py

```diff
@@ -0,0 +1,39 @@
+"""
+Common MongoDB connection functions.
+"""
+import pymongo
+from mongodb_proxy import MongoProxy
+
+
+# pylint: disable=unused-argument
+def connect_to_mongodb(db, collection, host,
+                       port=27017, tz_aware=True, user=None, password=None,
+                       retry_wait_time=0.1, proxy=True, **kwargs
+                       ):  # pylint: disable=bad-continuation
+    """
+    Returns a MongoDB Database connection, optionally wrapped in a proxy. The proxy
+    handles AutoReconnect errors by retrying read operations, since these exceptions
+    typically indicate a temporary step-down condition for MongoDB.
+    """
+    mongo_conn = pymongo.database.Database(
+        pymongo.MongoClient(
+            host=host,
+            port=port,
+            tz_aware=tz_aware,
+            document_class=dict,
+            **kwargs
```

Comment:
Would it make sense to just fold the named arguments into `**kwargs`, since they're all getting passed explicitly to `MongoClient`? I guess you'